R/optionalInput.R

```diff
@@ -302,7 +302,10 @@ updateOptionalSelectInput <- function(session, # nolint
     label = label,
     selected = as.character(raw_selected),
     choices = raw_choices,
-    choicesOpt = picker_options(choices)
+    choicesOpt = picker_options(choices),
+    options = list(
```

Comment:
in `optionalSelectInput` there is an options argument which is then compared with default options. If we want to keep options set during input initialization same thing should be done in update*.

```r
  default_options <- list(
    "actions-box" = multiple,
    "none-selected-text" = "- Nothing selected -",
    "max-options" = ifelse(multiple, Inf, 1),
    "show-subtext" = TRUE,
    "live-search" = ifelse(length(choices) > 10, TRUE, FALSE)
  )
  options <- modifyList(default_options, options)
  
  
```

```suggestion
    options = options
```

Reply:
The problem is we don't have `multiple` and `options` arguments/objects available in `updateOptionalSelectInput`.
https://github.com/insightsengineering/teal.widgets/blob/main/R/optionalInput.R#L290-L294

However, I think the `options`  from initializing `optionalSelectInput` get retain during `updateOptionalSelectInput` because I still see the `action-box` (Select/Deselect All) in the dropdown even though we don't specify this options in `updateOptionalSelectInput`.

<details>

<summary>Here's a simpler app to test the carryover of options setup</summary>

```r
library(shiny)
library(shinyWidgets)

ui <- fluidPage(
  shinyWidgets::pickerInput(
    inputId = "somevalue",
    label = "A label",
    choices = c("a", "b"),
    multiple = TRUE,
    options = list(
      "actions-box" = TRUE,
      "live-search" = TRUE
      
    )
  ),
  verbatimTextOutput("value")
)

server <- function(input, output) {
  
  observe({
    shinyWidgets::updatePickerInput(
      inputId = "somevalue",
      choices = c("a", "b", "c")
    )
  })
  
  output$value <- renderPrint(input$somevalue)
}

shinyApp(ui, server)
```

</details>

This is I only go for recomputing `live-search` option.

Reply:
Resolving the conversation based on this comment:
https://github.com/insightsengineering/teal.widgets/pull/292#pullrequestreview-2840098998